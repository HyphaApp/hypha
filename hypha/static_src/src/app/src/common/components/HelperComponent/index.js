import * as React from "react";
import PropTypes from 'prop-types';


const HelperComponent = props => {
  return <>
    {props.text && <p className="form__help" dangerouslySetInnerHTML={{ __html: props.text }} />}
    {props.link && <p className="form__help-link">
      <a href={props.link} target="_blank" rel="noopener noreferrer">See help guide for more information.
        <svg className="form__open-icon"><use href="#open-in-new-tab"></use></svg>
      </a>
    </p>}
  </>
}
HelperComponent.propTypes = {
  text: PropTypes.string,
  link: PropTypes.node
}

HelperComponent.displayName = 'HelperComponent';
export default HelperComponent;
 