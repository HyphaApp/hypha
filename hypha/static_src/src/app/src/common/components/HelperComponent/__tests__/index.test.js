import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import HelperComponent from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test Helper component", () => {
  const text = "test name";
  const link = "";

  const subject = mount(<HelperComponent
    text={text}
    link={link}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes", () => {
    expect(subject.find('.form__help').length).toBe(1);
    expect(subject.find('.form__help-link').length).toBe(0);
    expect(subject.children().length).toEqual(1)
  });

  test('render a Helper component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test Helper component with link", () => {
  const text = "test name";
  const link = "sdfsd";

  const subject = mount(<HelperComponent
    text={text}
    link={link}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes", () => {
    expect(subject.find('.form__help-link').length).toBe(1);
  });

  test('render a Helper component with link', () => {
    expect(subject).toMatchSnapshot();
  });

});
