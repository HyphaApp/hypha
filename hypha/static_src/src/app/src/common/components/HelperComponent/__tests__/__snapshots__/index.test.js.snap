// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Helper component render a Helper component 1`] = `
<HelperComponent
  link=""
  text="test name"
>
  <p
    className="form__help"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test name",
      }
    }
  />
</HelperComponent>
`;

exports[`Test Helper component with link render a Helper component with link 1`] = `
<HelperComponent
  link="sdfsd"
  text="test name"
>
  <p
    className="form__help"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test name",
      }
    }
  />
  <p
    className="form__help-link"
  >
    <a
      href="sdfsd"
      rel="noopener noreferrer"
      target="_blank"
    >
      See help guide for more information.
      <svg
        className="form__open-icon"
      >
        <use
          href="#open-in-new-tab"
        />
      </svg>
    </a>
  </p>
</HelperComponent>
`;
